torch/_export/serde/serialize.py

```diff
@@ -1528,6 +1526,15 @@ def deserialize_sym_int(self, s: SymInt) -> Union[int, torch.SymInt]:
     def deserialize_sym_bool(self, s: SymBool) -> Union[bool, torch.SymBool]:
         val = s.value
         if s.type == "as_expr":
+            # first we sympify this just to access any untracked symbols
+            expr = sympy.sympify(val.expr_str)
+            for sym in expr.free_symbols:
+                if (
+                    not isinstance(sym, sympy.Number)
+                    and str(sym) not in self.symbol_name_to_symbol
+                ):
+                    self.deserialize_sym_int(SymInt.create(as_expr=SymExpr(str(sym))))
+            # then we sympify again using locals to correctly reify with the constructed symbols
             expr = sympy.sympify(val.expr_str, locals=self.symbol_name_to_symbol)
```

Comment:
This is needed for SymBool expressions like `Eq(u0, 1)`, where u0 isn't bound to any previous node, and directly calling `create_symboolnode()` errors because we haven't tracked range info on u0, so we explicitly deserialize each symbol beforehand.

See `test_sym_bool` in test/export/test_serialize.py for example.